src/webhooks/helpers.ts

```diff
@@ -0,0 +1,60 @@
+// const { webcrypto, createHmac } = require("node:crypto");
+const { webcrypto } = require("node:crypto");
+const { subtle } = webcrypto;
+
+const ALGORITHM = { name: "HMAC", hash: "SHA-256" };
+
+const createHS256Signature = async ({ signingSecret, data }: { signingSecret: string; data: string }) => {
+    const enc = new TextEncoder();
+
+    let key = await subtle.importKey("raw", enc.encode(signingSecret), ALGORITHM, false, ["sign"]);
+    let signatureBuffer = await subtle.sign(ALGORITHM.name, key, enc.encode(data));
+    let signature = Buffer.from(signatureBuffer).toString("base64");
+
+    // Useful to check signature creation
+    // const hash = createHmac("sha256", signingSecret).update(data).digest("base64");
+    // console.log({ signature, hash });
+
+    return signature;
+};
+
+export type HookdeckVerifyArguments = {
+    signature: string;
+    secondarySignture?: string;
+    rawBody: string;
+    signingSecret: string;
+};
+
+export type HookdeckVerificationResult = {
+    isValidSignature: boolean;
+};
+
+/**
+ * Verify the Hookdeck webhook signature.
+ *
+ * @example
+ *     const result = await verifyWebhookSignature({
+ *         signature: "kKb0yljhY9tBo7oihOMTvRayKCUpCNVujkoTMNoGdGM=",
+ *         rawBody: `{"key1":"value1", "key2":"value2"}`,
+ *         signingSecret: "5fthvdmj8gtkzdv93mwkdpbupgwgfu4fu8xf4rro2oufn6qlhc",
+ *     })
+ *
+ *     if(!result.isValidSignature) {
+ *         // Reject the webhook payload
+ *     }
+ *     else {
+ *         // Proceed with the webhook payload
+ *     }
+ */
+export const verifyWebhookSignature = async ({
+    signature,
+    rawBody,
+    signingSecret,
+    secondarySignture,
```

Comment:
Typo

Reply:
👍 fixed in 5d33f9592a72e759528a716f941705be040e1b50